pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/tasks/Flake8Task.groovy

```diff
@@ -21,14 +21,46 @@ import org.gradle.process.ExecResult
 @CompileStatic
 public class Flake8Task extends AbstractPythonMainSourceDefaultTask {
 
-  public void preExecution() {
-    args(pythonDetails.virtualEnvironment.findExecutable("flake8").absolutePath,
-        "--config", "$component.setupCfg",
-        "$component.srcDir",
-        "$component.testDir")
-  }
-
-  @Override
-  void processResults(ExecResult execResult) {
-  }
+    public void preExecution() {
+        /*
+         Modified to only include folders that exist. if no folders exist, then
+         the task isn't actually run.
+          */
+        List<String> sArgs = [pythonDetails.virtualEnvironment.findExecutable("flake8").absolutePath,
+                     "--config", component.setupCfg]
+
+        if (!project.file(pythonDetails.virtualEnvironment.findExecutable("flake8").absolutePath).exists()) {
+            project.logger.error("Flake8 executable not found")
+            project.logger.error("Flake8: ${pythonDetails.virtualEnvironment.findExecutable("flake8").absolutePath}")
+        } else {
+            project.logger.info("Flake8: found executable")
+        }
+
+        def paths = []
+        if (project.file(component.srcDir).exists()) {
+            project.logger.info("Flake8: adding ${component.srcDir}")
+            paths.add(component.srcDir)
+        } else {
+            project.logger.info("Flake8: srcDir doesn't exist")
+        }
+
+        if (project.file(component.testDir).exists()) {
+            project.logger.info("Flake8: adding ${component.testDir}")
+            paths.add(component.testDir)
+        } else {
+            project.logger.info("Flake8: testDir doesn't exist")
+        }
+
+        if (paths.size() > 0) {
```

Comment:
Done, but there was a complication.  Narc complained that the PythonPluggins class went over 350 lines (352 to be exact).  So to make it smaller, I tried moving the constants to an interface.  Been doing that for 15 years now.  Apparently, thats not considered a good design pattern anymore cuz Narc complained about that.  Turns out, for big lists of constants like that the prefered pattern is an Enum.  So i had to convert everything to an enum and update the codebase.  When I do the Sphinx conversion, I will split this file up into sub classes.  I did it with another plugin I have, its more planning than work.  When i do that, i can convert things back to string constants if you want because i will have the room in the file.